#150-create edit project functionality

ALau2088 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

#### Issue addressed
 include `fixes #150 ` 
<!-- the relevant issue in https://github.com/AgileVentures/WebsiteOne-FE/issues  -->

fixes #

#### Screenshots (if appropriate):
<!-- please include screenshots of any changes to the UI for quick review  -->
<!-- please show how things look on both desktop and mobile  -->

#### Testing
<!-- Remember you must see any new tests you created (or old ones you changed) -->
<!-- fail as well as pass in order to ensure they are working -->
<!-- Unsure how to test? - please see https://github.com/AgileVentures/WebsiteOne-FE/blob/develop/CONTRIBUTING.md#git-and-github-->

ALau2088 committed:
create edit project functionality

ALau2088 committed:
150-edit_project

mattwr18 commented:
hey thanks @ALau2088 
great work getting this in!! I'll have a closer look and give you a review soon, I have a pretty good idea of the code since we worked on it in the `React Mob`, but I'll look closer.

A couple things blocking it from a thorough review are that it's failing the build cause of linting issues. Could you run `yarn fix` and push it back up? Also, if you run the unit tests(jest) locally before pushing up, it'll run `standard` for you and you'll see the linting errors quicker. 

That's also good practice because sometimes we don't expect our changes to break other tests, but they do sometimes and you'll catch those faster as well.

Also, please `rebase` :smile_cat: 
Looking forward to getting this merged!!

ALau2088 committed:
resolve merge conflicts

ALau2088 committed:
150-edit_project

mattwr18 commented:
hey @ALau2088, you might have seen already(?), but it's failing the build cause a few of the tests this `PR` introduces are failing. Are they failing in the same way locally?

Let me know if you want to have a pairing session to look at them together.

ALau2088 committed:
150-edit_project

ALau2088 committed:
150-edit_project

ALau2088 committed:
150-edit_project

ALau2088 committed:
150-edit_project

ALau2088 committed:
150-edit_project

ALau2088 committed:
150-edit_project

mattwr18 commented:
when can we schedule a time to work on this together @ALau2088, unfortunately the last time I had a power outage :worried:

ALau2088 commented:
Hey @mattwr18 . I am free to pair Monday 8am-12pm PST.

mattwr18 commented:
great let's do it at `5pm UTC` @ALau2088

ALau2088 commented:
@mattwr18 .Cool. See you then.

ALau2088 committed:
rebase upstream/develop

ALau2088 committed:
rebase upstream/develop

ALau2088 committed:
refactor history.push and tests

ALau2088 commented:
@mattwr18 I refactored editProjectPage, editProjectAction and their respective unit tests. It is passing now.

For the Cypress test. Do I I just modify ~cypress/integration/common/web_steps.js ?

mattwr18 commented on src/containers/EditProjectPage.js:
```diff
@@ -0,0 +1,74 @@
+      cookies,
+      history
+    })
+    history.push(`/projects/${this.props.match.params.slug}`)
```
I think we want this to be in a callback @ALau2088 so that it pushes the user only if the `axios` request resolved, no?

mattwr18 reviewed (changes requested):
I think that you can add the tests to `cypress/integration`, maybe you can rename the `create_project.feature` to `project.feature` and have another scenario that is for editing a project?
Then, yes I think you can add the steps to the `web_steps`... at least for now this is how we have it organized, we can always have the discussion of refactoring it to have multiple shorter files organized by feature, for example....
It's looking good!!

mattwr18 commented on src/actions/editProjectAction.js:
```diff
@@ -0,0 +1,28 @@
+      Accept: 'application/json'
+    }
+  })
+    .then(response => console.log(response))
```
please remove this callback, as we're not using it. Also, the test coverage is dropping/checks are failing cause it says we're not testing this line

ALau2088 committed:
150-edit_project

ALau2088 committed:
150-edit_project

ALau2088 committed:
work on cypress test

ALau2088 committed:
Merge branch 'develop' of https://github.com/AgileVentures/WebsiteOne-FE into 150-edit_project

ALau2088 committed:
add edit button to project info page

ALau2088 committed:
update cypress test

ALau2088 committed:
add edit_project.feature and cypress tests

ALau2088 commented:
@mattwr18 , I pushed the edit_project.feature and the web_steps tests for edit project page. The connection to the server is choppy and I am having difficulty running it. Maybe you can run it and let me know the results?

aonomike commented:
@ALau2088 I think the build is failing because of linting issue, maybe you want to run `yarn lint --fix`?

ALau2088 committed:
add ProjectSummary.js to standard ignore

ALau2088 commented:
@aonomike . That's correct. It keeps saying ProjectSummary.js has space linting issues but it seems find to me. For now, I just add ProjectSummary.js to standard ignore files list in package.json to get it running.

ALau2088 commented:
@aonomike @mattwr18 I am currently dealing with this issue with edit_project.feature. Any thoughts?
<img width="960" alt="9" src="https://user-images.githubusercontent.com/37641224/60834855-84ef5100-a176-11e9-9139-ec6ba87c388c.png">

aonomike commented:
> @aonomike . That's correct. It keeps saying ProjectSummary.js has space linting issues but it seems find to me. For now, I just add ProjectSummary.js to standard ignore files list in package.json to get it running.

Yes @clementkng and I were both having trouble with ProjectSummary.js and standard as well. Maybe its good to ignore it for now...Thanks!!

aonomike committed:
Resolve failing tests

- Clear all mocks after each test
- Update the test for fetchProjectAction call

Co-authored-by: Clement Ng <clementdng@gmail.com>

aonomike committed:
Update cypress tests

- Use mocks in place of direct api access
- We are kinda stuck with the test for the update action since the codebase doesnt have an 'EDIT_PROJECT' Action type to make the updates. @mattwr18 we may need help with this.

Co-authored-by: Clement Ng <clementdng@gmail.com>

aonomike commented:
Hi @ALau2088 we worked at your PR together with @clementkng and pushed our work to [this branch](https://github.com/aonomike/WebsiteOne-FE/tree/150-edit_project) We resolved failing jest tests and your cypress tests. We did not push to your remote since you were not aware when we were making changes. Kindly review and let us know if we can just go ahead and push. Thanks!!

ALau2088 commented:
@aonomike Looks good to me. Thanks.

aonomike committed:
Fix linting errors

aonomike commented:
Just pushed up the changes to this branch @ALau2088
